CSharpSourceCode/Battle/Map/TowMapWeatherModel.cs

```diff
@@ -0,0 +1,198 @@
+﻿using NLog;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml;
+using TaleWorlds.CampaignSystem;
+using TaleWorlds.CampaignSystem.SandBox.GameComponents.Map;
+using TaleWorlds.Library;
+using TaleWorlds.ModuleManager;
+using TOW_Core.Utilities;
+using TOW_Core.Utilities.Extensions;
+
+namespace TOW_Core.Battle.Map
+{
+    public class TowMapWeatherModel : DefaultMapWeatherModel
+    {
+        private readonly string ModuleName = "TOW_EnvironmentAssets";
+        private readonly string ForceAtmosphereKey = "forceatmo";
+
+        //Season codes from TW code: https://imgur.com/p5CtVt0
+        private readonly Dictionary<string, int> SeasonCodes = new Dictionary<string, int>()
+        {
+            {"spring",  0},
+            {"summer", 1},
+            {"fall", 2},
+            {"winter", 3}
+        };
+
+        public override AtmosphereInfo GetAtmosphereModel(CampaignTime timeOfYear, Vec3 pos)
```

Comment:
I work with t33baum on his Time model. I don't know yet but I hope that we can have in later stages our own CampaignTime System.  Has nothing todo with this here.